subsys/net/lib/download_client/src/http_download_client.c

```diff
@@ -49,71 +56,126 @@ static int socket_timeout_set(int fd)
 	return 0;
 }
 
-static int resolve_and_connect(const char *const host, const char *const port,
-			       u32_t family, u32_t proto)
+static int socket_sectag_set(int fd, int sec_tag)
+{
+	int err;
+	int verify;
+	sec_tag_t sec_tag_list[] = { sec_tag };
+
+	enum {
+		NONE = 0,
+		OPTIONAL = 1,
+		REQUIRED = 2,
+	};
+
+	verify = OPTIONAL;
+
+	err = setsockopt(fd, SOL_TLS, TLS_PEER_VERIFY, &verify, sizeof(verify));
+	if (err) {
+		LOG_ERR("Failed to setup peer verification, errno %d", errno);
+		return -1;
+	}
+
+	err = setsockopt(fd, SOL_TLS, TLS_SEC_TAG_LIST, sec_tag_list,
+			 sizeof(sec_tag_t) * ARRAY_SIZE(sec_tag_list));
+	if (err) {
+		LOG_ERR("Failed to setup socket security tag, errno %d", errno);
+		return -1;
+	}
+	return 0;
+}
+
+static int resolve_and_connect(const char *const host, int family, int sec_tag)
 {
 	int fd;
-	int rc;
+	int err;
+	int proto;
+	u16_t port;
+	struct addrinfo *addr;
+	struct addrinfo *info;
 
 	if (host == NULL) {
 		return -1;
 	}
 
-	struct addrinfo *addrinf = NULL;
+	/* Set up port and protocol */
+	if (sec_tag < 0) {
```

Comment:
Why don't we check explicitly with -1?

Reply:
No reason in particular, changed to check against `-1`.

Reply:
alright